libs/src/signed_integers/common.sw

```diff
@@ -1,11 +1,15 @@
 library;
 
-/// Trait for the Two's Complement of a value.
-pub trait TwosComplement {
-    /// Returns the two's complement of a value.
+/// Wrapping (modular) negation. Computes -self, wrapping around at the boundary of the type.
+pub trait WrappingNeg {
+    /// Negates a signed number.
+    ///
+    /// # Additional Information
+    ///
+    /// * The only case where such wrapping can occur is when one negates self::min(). In such a case, this function returns self::min() itself.
```

Comment:
Not sure i understand this, shouldnt -(self::min) return self::max?

Reply:
Trying 

`fn main () {
    println!("{}", -i8::MIN)
}`
On returns a compiler error. Not sure if returning self.min is the right behavior here, would prefer a result or a revert

Reply:
This is correct behaviour for [wrapping_neg](https://doc.rust-lang.org/std/primitive.i32.html#method.wrapping_neg).

Although I agree that [checked_neg](https://doc.rust-lang.org/std/primitive.i32.html#method.checked_neg) is less user error-prone.

Reply:
No max value is 2 ^ 7 - 1, min value is - 2 ^ 7